app/src/userpages/components/Header/Search/index.jsx

```diff
@@ -0,0 +1,97 @@
+// @flow
+
+import React, { useState, useEffect, useCallback, useRef } from 'react'
+import { type Ref } from '$shared/flowtype/common-types'
+import debounce from 'lodash/debounce'
+import cx from 'classnames'
+import SvgIcon from '$shared/components/SvgIcon'
+import styles from './search.pcss'
+
+type Props = {
+    onChange?: ?(string) => void,
+    placeholder?: ?string,
+    value: ?string,
+}
+
+const Search = ({ value: valueProp, onChange: onChangeProp, placeholder }: Props) => {
+    const [value, setValue] = useState(valueProp || '')
+
+    const [focused, setFocused] = useState(false)
+
+    const open = !!(focused || value)
+
+    useEffect(() => {
+        setValue(valueProp || '')
+    }, [valueProp])
+
+    const onFocus = useCallback(() => {
+        setFocused(true)
+    }, [])
+
+    const onBlur = useCallback(() => {
+        setFocused(false)
+    }, [])
+
+    const debouncedOnChange = useCallback(debounce((value: string) => {
```

Comment:
Note that if/when this creates a new `debouncedOnChange`, the old `debouncedOnChange` won't magically disappear and will still fire after the delay if it's currently pending.

Reply:
I literally was dreaming about someone pointing that out to me today. There you go. I'll add a check.

Reply:
@timoxley can you check it out now? I implemented it so that all pending calls to previous debounces are cancelled. Let me know what you think.